File: solutions/lua/leap/1/leap.lua
local leap_year = function(number)

    -- a leap year is evenly divisible by 4
    -- but not that is evenly divisible by 100
    -- unless it is is also evenly divisible by 400
    return (number % 4 == 0) and ((number %100 ~= 0) or (number % 400 == 0))
end

return leap_year

File: solutions/c/leap/1/leap.c
#include "leap.h"

bool is_divisible_by(int n, int m);

bool leap_year(int year) {

    return is_divisible_by(year, 4) 
        && (!is_divisible_by(year, 100) || is_divisible_by(year, 400));
}

bool is_divisible_by(int n, int m) {
    return (n % m) == 0;
}